### PR TITLE
Upgrading SB pkg and adding multiple factories

### PR DESCRIPTION
**Pull request by rajeevmv**
- Upgrading SB pkg
- adding multiple factories
- disable sender client side batching

This is just a merge into my own private fork. Just so that I can easily send out a CF.

Reference: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-performance-improvements

### CODE REVIEW

**Comment by rajeevmv** [comment]
@alextolp is added to the review. #Closed

**Comment by rajeevmv** [comment]
@Simonp is added to the review. #Closed

**Comment by rajeevmv** [comment]
@simonporter is added to the review. #Closed

File: src/DurableTask.ServiceBus/ServiceBusOrchestrationService.cs [hunk 1]
```diff
@@ -148,12 +161,18 @@ public class ServiceBusOrchestrationService : IOrchestrationService, IOrchestrat
...
+            this.trackingQueueClientMessagingFactory = ServiceBusUtils.CreateReceiverMessagingFactory(namespaceManager, sbConnectionStringBuilder, trackingEntityName);
+
+            orchestratorSender = ServiceBusUtils.CreateMessageSender(this.orchestratorSenderMessagingFactory, orchestratorEntityName, workerEntityName);
+            workerSender = ServiceBusUtils.CreateMessageSender(this.workerSenderMessagingFactory, workerEntityName, orchestratorEntityName);
+            trackingSender = ServiceBusUtils.CreateMessageSender(this.trackingSenderMessagingFactory, trackingEntityName, orchestratorEntityName);
+            workerQueueClient = ServiceBusUtils.CreateQueueClient(this.workerQueueClientMessagingFactory, workerEntityName);
+            orchestratorQueueClient = ServiceBusUtils.CreateQueueClient(this.orchestratorQueueClientMessagingFactory, orchestratorEntityName);
+            trackingQueueClient = ServiceBusUtils.CreateQueueClient(this.trackingQueueClientMessagingFactory, trackingEntityName);
             if (trackingDispatcher != null)
```
Comment (alextolp) [inline src/DurableTask.ServiceBus/ServiceBusOrchestrationService.cs]: nit: new line #Resolved

File: src/DurableTask.ServiceBus/Common/ServiceBusUtils.cs [hunk 2]
```diff
@@ -313,5 +314,69 @@ public static MessagingFactory CreateMessagingFactory(string connectionString)
...
+        {
+            MessagingFactory messagingFactory = MessagingFactory.Create(
+                namespaceManager.Address.ToString(),
+                new MessagingFactorySettings
+                {
+                    TransportType = TransportType.NetMessaging,
+                    TokenProvider = TokenProvider.CreateSharedAccessSignatureTokenProvider(
+                        sbConnectionStringBuilder.SharedAccessKeyName, 
+                        sbConnectionStringBuilder.SharedAccessKey, 
+                        TimeSpan.FromDays(365)),
```
Comment (alextolp) [inline src/DurableTask.ServiceBus/Common/ServiceBusUtils.cs]: >365 [](start = 42, length = 3)

perhaps shorter ttl? #Resolved

File: src/DurableTask.ServiceBus/Common/ServiceBusUtils.cs [hunk 3]
```diff
@@ -313,5 +314,69 @@ public static MessagingFactory CreateMessagingFactory(string connectionString)
...
+
+            MessagingFactory messagingFactory = MessagingFactory.Create(
+                namespaceManager.Address.ToString(),
+                new MessagingFactorySettings
+                {
+                    TransportType = TransportType.NetMessaging,
+                    TokenProvider = TokenProvider.CreateSharedAccessSignatureTokenProvider(
+                        sbConnectionStringBuilder.SharedAccessKeyName, 
+                        sbConnectionStringBuilder.SharedAccessKey, 
+                        TimeSpan.FromDays(365))
```
Comment (alextolp) [inline src/DurableTask.ServiceBus/Common/ServiceBusUtils.cs]: >365 [](start = 42, length = 3)

same #Resolved

**Review by alextolp — APPROVED** [review]
:shipit:

**Pull request merged**